File: option.py
import sys
import os
import math


__name__ == "__main__"


class index(float) :
    def scan():
        sc = input("")
        return sc    

class sum(index):

 def main():
  while(True):
    print("enter numbers which you want to add :~~")
    a = index.scan()
    print("+")
    b = index.scan()
    c = float(a) + float(b)
    print("=   " , c)
    
    print("do you want to quit or you want to go to the menu(if yes enter 'y' else enter 'n' )" + "(for menu enter m)")
    str = index.scan()
    if str == "y":
      print("Bye :)")
      exit(0)

    elif str == "n":
        continue

    elif str == "m":
       os.execv(sys.executable, ['python3'] + sys.argv)
       
     

    else:
        print("Wrong Input :(")
        print("try again :)")
        exit(0)


class diff(index):

 def main():
  while(True):
    print("enter numbers which you want to subtract :~~")
    a = index.scan()
    print("-")
    b = index.scan()
    c = float(a) - float(b)
    print("=   " , c)
    
    print("do you want to quit or you want to go to the menu(if yes enter 'y' else enter 'n' )" + "(for menu enter m)")
    str = index.scan()
    if str == "y":
      exit(0)

    elif str == "n":
        continue

    elif str == "m":
       os.execv(sys.executable, ['python3'] + sys.argv)
       

    else:
        print("Wrong Input :(")
        print("try again :)")
        exit(0)


class divide(index):

 def main():
  while(True):
    print("enter the two want to divide billow :~~")
    a = index.scan()
    print("/")
    b = index.scan()
    c = float(a) / float(b)
    print("=   " , c)
    
    print("do you want to quit or you want to go to the menu(if yes enter 'y' else enter 'n' )" + "(for menu enter m)")
    str = index.scan()
    if str == "y":
      print("Bye :)")
      exit(0)

    elif str == "n":
        continue

    elif str == "m":
       os.execv(sys.executable, ['python3'] + sys.argv)
       
     

    else:
        print("Wrong Input :(")
        print("try again :)")
        exit(0)


class multiply(index):

 def main():
  while(True):
    print("enter the number want to multiply billow :~~")
    a = index.scan()
    print("x")
    b = index.scan()
    c = float(a) * float(b)
    print("=   " , c)
    
    print("do you want to quit or you want to go to the menu(if yes enter 'y' else enter 'n' )" + "(for menu enter m)")
    str = index.scan()
    if str == "y":
      print("Bye :)")
      exit(0)

    elif str == "n":
        continue

    elif str == "m":
       os.execv(sys.executable, ['python3'] + sys.argv)
       
     

    else:
        print("Wrong Input :(")
        print("try again :)")
        exit(0)


class evenOdd(index):

 def main():
  while(True):
    print("enter the number you want to check :~~ ")
    a = int(index.scan())
    if (a % 2) == 0 :
       print(a, " is even")
    else :
       print(a, " is odd")
    
    print("do you want to quit or you want to go to the menu(if yes enter 'y' else enter 'n' )" + "(for menu enter m)")
    str = index.scan()
    if str == "y":
      print("Bye :)")
      exit(0)

    elif str == "n":
        continue

    elif str == "m":
       os.execv(sys.executable, ['python3'] + sys.argv)
       
     

    else:
        print("Wrong Input :(")
        print("try again :)")
        exit(0)


class factorial(index):

 def main():
  while(True):
    print("enter the number want to calculate the factorial of :~~")
    a = int(index.scan())
    facto = math.factorial(a)
    print("factorial of ",a,"=  " , facto)
    print("do you want to quit or you want to go to the menu(if yes enter 'y' else enter 'n' )" + "(for menu enter m)")
    str = index.scan()
    if str == "y":
      print("Bye :)")
      exit(0)

    elif str == "n":
        continue

    elif str == "m":
       os.execv(sys.executable, ['python3'] + sys.argv)
       
     

    else:
        print("Wrong Input :(")
        print("try again :)")
        exit(0)


class option(index):
 
 def option():
    print("lets do some calculations")
    print("what do you want to do ")
    print("Here are your options")
    print("0- Exit")
    print("1- Sum")
    print("2- Difference")
    print("3- Divide")
    print("4- Multiply")
    print("5- Check even or odd")
    print("6- To calculate factorial")
    print("enter any number listed above to continue :)")

    a = index.scan()

    while(True):
        if a == "0":
            print("Bye :) ")
            exit(0)
            
        elif a == "1":
             print("ok, good choice. Lets do some additions :)")
             exec(sum.main())

            
        elif a == "2":
             print("ok, good choice lets find some difference  ;) ")
             exec(diff.main())

        elif a == "3":
             print("ok, good choice lets do some divisions  :] ")
             exec(divide.main())

        elif a == "4":
             print("ok, good choice lets do some multiplication :) ")
             exec(multiply.main())

        elif a == "5":
             print("lets check even odd")
             exec(evenOdd.main())

        elif a == "6":
             print("Factorials ! cool !")
             exec(factorial.main())

        else:
             print("Invalid Input :(")
             os.execv(sys.executable, ['python3'] + sys.argv)
            
                

        
 option()   
       

        
            
            

File: option.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

float scan(){
    float num;
    scanf("%f", &num);
    return num;
}

char scanc(){
    char choice;
    printf("Do you want to continue ? \n");
    printf("---- ---- ---- ---- \n");
    printf("y = Yes \n");
    printf("n = Exit \n");
    printf("m = Return to Menu \n");
    printf("---- ---- ---- ---- \n");
    scanf(" %c", &choice);
    return choice;
}

char err(){
  printf("Wrong Input :()\n Try again :) \n");
}

float plus(){
    while(true){

        float a, b, c;
        printf("Enter Number you want to add \n");
        a = scan();
        printf("+ \n");
        b = scan();
        c = a + b;
        printf("Answer of ans is %g \n", c);
        return c;

        
      }

}

float minus(){
    while(true){

        float a, b, c;
        printf("Enter Number you want to minus from \n");
        a = scan();
        printf("- \n");
        b = scan();
        c = a - b;
        printf("Answer of ans is %g \n", c);
        return c;

        
        
      }

}

float divide(){
    while(true){

        float a, b, c;
        printf("Enter Number you want to divide \n");
        a = scan();
        printf("/ \n");
        b = scan();
        c = a / b;
        printf("Answer of ans is %g \n", c);
        return c;

        
      }

}

float multiply(){
    while(true){

        float a, b, c;
        printf("Enter Number you want to multiply \n");
        a = scan();
        printf("x \n");
        b = scan();
        c = a * b;
        printf("Answer of ans is %g \n", c);
        return c;

      
    }

}

float evenodd(){
  while(true){
    
     int n;
     printf("Enter an integer: ");
     scanf("%d", &n);
     (n % 2 == 0) ? printf("%d is even. \n", n) : printf("%d is odd. \n", n);
     return n;
  }
}

float factorial(){
  while(true){
    int n, i;
    unsigned long long fact = 1;
    printf("Enter an integer: ");
    scanf("%d", &n);
    if (n < 0)
        printf("Error! Factorial of a negative number doesn't exist. \n");
    else {
        for (i = 1; i <= n; ++i) {
            fact *= i;
        }
        printf("Factorial of %d = %llu \n", n, fact);
    }

    return fact;
  }
}


int options(){
    printf("lets do some calculations \n");
    printf("what do you want to do \n");
    printf("Here are your options \n");
    printf("0- Exit \n");
    printf("1- Sum \n");
    printf("2- Difference \n");
    printf("3- Divide \n");
    printf("4- Multiply \n");
    printf("5- Check even or odd \n");
    printf("6- To calculate factorial \n");
    printf("enter any number listed above to continue :) \n");

    float value;
    value = scan();
    return value;


}

int main(){
    int choice = options();
    while(true){

       if(choice == 0){
         printf("Bye :)");
         exit(0);
       }
       if(choice == 1){
          while(true){
             plus();
             char d;
             d = scanc();
              if(d == 'y'){
                continue;
              }
              if(d == 'n'){
                printf("Bye :)");
                exit(0);
              }
              if(d == 'm'){
                main();
              }
              else{
                err();
                continue;
              }
            }
            
         
        }
       if(choice == 2){
        while(true){
             minus();
             char d;
             d = scanc();
              if(d == 'y'){
                continue;
              }
              if(d == 'n'){
                printf("Bye :)");
                exit(0);
              }
              if(d == 'm'){
                main();
              }
              else{
                err();
                continue;
              }
            }
        }
        if(choice == 3){
        while(true){
             divide();
             char d;
             d = scanc();
              if(d == 'y'){
                continue;
              }
              if(d == 'n'){
                printf("Bye :)");
                exit(0);
              }
              if(d == 'm'){
                main();
              }
              else{
                err();
                continue;
              }
            }

        }
        if(choice == 4){
        while(true){
             multiply();
             char d;
             d = scanc();
              if(d == 'y'){
                continue;
              }
              if(d == 'n'){
                printf("Bye :)");
                exit(0);
              }
              if(d == 'm'){
                main();
              }
              else{
                err();
                continue;
              }
            }

        }
        if(choice == 5){
        while(true){
             evenodd();
             char d;
             d = scanc();
              if(d == 'y'){
                continue;
              }
              if(d == 'n'){
                printf("Bye :)");
                exit(0);
              }
              if(d == 'm'){
                main();
              }
              else{
                err();
                continue;
              }
            }

        }
        if(choice == 6){
        while(true){
             factorial();
             char d;
             d = scanc();
              if(d == 'y'){
                continue;
              }
              if(d == 'n'){
                printf("Bye :)");
                exit(0);
              }
              if(d == 'm'){
                main();
              }
              else{
                err();
                continue;
              }
            }

        }
        else{
         err();
         while(true){
             char d;
             printf("Retry ? \n");
             printf("--- --- \n");
             printf("y = yes \n");
             printf("n = no \n");
             printf("--- --- \n");
             scanf(" %c", &d);
             d = scanc();
              if(d == 'y'){
                main();
              }
              if(d == 'n'){
                printf("Bye :)");
                exit(0);
              }
              else{
                err();
                continue;
              }
            }

        }
    }

}